src/help/git.ts

```diff
@@ -7,17 +7,61 @@ import * as shell from './shell';
  * @param targetDir the clone directory.
  */
 export function clone(repositoryUrl: string, targetDir: string) {
-  const gitHubUseSsh = process.env.GITHUB_USE_SSH;
+  const gitHubUseSsh = detectSSH();
   if (gitHubUseSsh) {
     const sshRepositoryUrl = repositoryUrl.replace('/', ':');
     shell.run(`git clone git@${sshRepositoryUrl}.git ${targetDir}`);
   } else {
-    const gitHubToken = process.env.GITHUB_TOKEN;
+    const gitHubToken = getToken(detectGHE());
     if (!gitHubToken) {
       throw new Error('GITHUB_TOKEN env variable is required when GITHUB_USE_SSH env variable is not used');
     }
     shell.run(`git clone https://${gitHubToken}@${repositoryUrl}.git ${targetDir}`);
+
+  }
+}
+
+/**
+ * Checks if the GitHub API URL set in the GitHub Actions workflow is not the public GitHub API URL.
+ * https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
+ *
+ * @return True if GITHUB_API_URL env var is defined and not equal to public github API URL, false,
```

Comment:
```suggestion
 * Checks if the current environment is an GHE environment.
 *
 * This check is using GITHUB_API_URL set in GitHub Actions workflow, as well as common gh cli env variables.
 * https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
 * https://cli.github.com/manual/gh_help_environment
 *
 * @return - `true` if GH_HOST or GITHUB_API_URL env var are defined and not equal to the public github endpoint, otherwise `false`
```